### PR TITLE
[-] CORE : actually use Context->shop in Link->getLanguageLink()

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by vTerenti** [comment]
(note: this is an automated message, but answering it will reach a real human )

Hi! Thank you for your pull request: your contribution to PrestaShop means a lot to us!

We are aware that this PR is still open after all this time, and we want to make sure that your hard work is not forgotten.

We are diving back into pull requests, and we want to work with you in order to get your PR to completion: we need your feedback!<br/>
Could you read [this article](http://build.prestashop.com/news/review-your-pull-requests/) from [the Build devblog](http://build.prestashop.com/), and take the time to review your pull requests, so that we can work with up-to-date information? You don't have to do it all right now, and every little step helps.

Again, thank you very much for this contribution, and for your renewed feedback on it!

**Comment by ZiZuu-store** [comment]
So .. this is rejected or just closed because nobody want to loose his time to work on it?  ... Just for the sake of knowledge :)

**Comment by Quetzacoalt91** [comment]
Hello,

This PR has been automatically closed because the branch `1.6` does not exist anymore.

Best regards

**Comment by ZiZuu-store** [comment]
It could be cherry picked ... Am I wrong?

We are contributing our little 2 cents whenever possible, but our main goal is our online shop. If bug fixes or improvements we share will die this way then will have to use better out time in future ;(

I don't want to be boring or not polite .. but please understand that contributors like us do not develop for business, so our little patch is a great effort and it is frustrating to see a "real life issue we already solved and contributed bak to the community" being still there because we had no time to re-write or re-contribute ... I think that *if the contribution is good* it should be a PS core developing team task to assure the problem is eradicated once and forever in the code 

Sorry for the long, bad english, flow of words :)

**Comment by julienbourdeau** [comment]
Hi,

Reviewing and accepting PRs is not so easy. We have to check that it has no implications, it means that if it causes bugs later **we** have to maintain this code. In your PR you provide no description about what the problem is and why it's fixing anything.

I understand how you feel but we closed this one because it was old and on the former `1.6` branch. I'm not sure this will cherry-pick nicely, so we'll have to report the change manually.

I'm sorry but we have a lot to do with the upcoming PrestaShop 1.7 and we try to work as fast and as efficiently as possible with the new PRs on develop, unfortunately there is not much time left for old PRs.

**Comment by ZiZuu-store** [comment]
If you look to the code you see easily that it simply passes missing arguments to core functions ..  I understand you, but you should understand me too, as I'm not responsible for the patch being "old" :)

If you promise to take it into account ASAvP I'll find the time to redo it against dev branch if you want ..

**Comment by xBorderie** [comment]
@ZiZuu-store If you can redo it on the 1.6.1.x or develop branch, with full details and correct commit names, I'm pretty sure the PR can be taken into account much faster now :)
In any case, the 1.6 branch itself was closed, so it's not a matter of judging the interest of your changes: we wouldn't have been able to merge into the current codebase anyway :/

**Comment by ZiZuu-store** [comment]
eheh @xBorderie  at the time of the PR (if I'm not wrong) there was no PR template ;)

I'll do .. ASAP Thanks

**Comment by kpodemski** [comment]
@ZiZuu-store to be honest for all this years it was like that and everything was ok, is there any real advantage of this PR?

**Comment by xBorderie** [comment]
Please note that I make not promise that your PR will be merged, only that you should receive a much faster response with an up-to-date PR :)

**Comment by julienbourdeau** [comment]
@ZiZuu-store The parameters are **not** missing. All across PrestaShop, the point of passing the `$id_shop` is to use a different one from the context.

I checked, all these method use `getBaseLink` which use the id_shop from the context if null. 
Thats why I need to get details on the problem you're trying to fix. Have you encountered errors?

**Comment by xBorderie** [comment]
Details... and a step by step way to reproduce the issue every time -- and thus to test if the PR does fix it in all cases :)

**Comment by ZiZuu-store** [comment]
Will  provide, sure

Of course my use case was to get the language link for different shops..
Il 28/giu/2016 19:04, "Xavier" <notifications@github.com> ha scritto:

Details... and a step by step way to reproduce the issue every time -- and
thus to test if the PR does fix it in all cases :)

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/PrestaShop/PrestaShop/pull/2958#issuecomment-229114160>,
or mute the thread
<https://github.com/notifications/unsubscribe/AFvVf60SDqbFAYjr230uCkMxTjrsQ9OZks5qQVQPgaJpZM4ELyiA>
.

**Comment by ZiZuu-store** [comment]
@julienbourdeau of course I was using it to get data about **different shops**, not the actual one being in the context

File: classes/Link.php [hunk 1]
```diff
@@ -499,29 +499,31 @@ public function getLanguageLink($id_lang, Context $context = null)
 		if (!empty($context->controller->php_self))
 			$controller = $context->controller->php_self;
 
+		$id_shop = (int)$context->shop->id;
```
Comment (ZiZuu-store) [inline classes/Link.php]: @julienbourdeau the point is HERE ... The method accept a Context, but does not fully honor it ...

**Comment by ZiZuu-store** [comment]
So .. as the method accept a Context parameter it is **really expected** that **if** invoked with a given Context parameter (that could be different from the actual one, I could make a new Context instance, change the shop, the language and so on) it would take its info from the passed object and **not** from the "global" Context actually in the controller ...

The patch is IMHO independent from the specific use case .. it is a fix of a wrong behavior :)

**Comment by kpodemski** [comment]
You're right, $id_shop is never taked into account in this method, go for rebase or new PR : )

**Comment by xBorderie** [comment]
New PR, since this one was made on a now-closed branch :)

**Comment by ZiZuu-store** [comment]
Puff puff .. how hard to be understood ;) 

Will make a new, better, PR .. stay tuned guys :)

**Pull request closed without merging**